coordinator/src/test/scala/filodb.coordinator/queryplanner/PlannerHierarchySpec.scala

```diff
@@ -308,4 +308,104 @@ class PlannerHierarchySpec extends AnyFunSpec with Matchers with PlanValidationS
+       |-E~LabelCardinalityReduceExec() on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-93033340],raw)
+       |--E~LabelCardinalityExec(shard=0, filters=List(ColumnFilter(_ws_,Equals(ws)), ColumnFilter(_ns_,Equals(ns)), ColumnFilter(_metric_,Equals(metric))), limit=1000000, startMs=1633913330000, endMs=1634777330000) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-93033340],raw)
+       |--E~LabelCardinalityExec(shard=1, filters=List(ColumnFilter(_ws_,Equals(ws)), ColumnFilter(_ns_,Equals(ns)), ColumnFilter(_metric_,Equals(metric))), limit=1000000, startMs=1633913330000, endMs=1634777330000) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-93033340],raw)""".stripMargin
+    validatePlan(rawPlan, expectedRawPlannerPlan)
+
+    // LTRPlanner cases
+    // Case 1: startTime < rawStart < dsStartTime < endTime
+    // Should go to both clusters
+    // DS Time range should be [userStartTime, dsEndTime]
+    // Raw Time range should be [rawStartTime, userEndTime]
+
+    val longTermPlan = rootPlanner.materialize(logicalPlan, queryContext)
+
+    val expectedLongTermPlan =
+           """T~LabelCardinalityPresenter(LabelCardinalityPresenter)
+             |-E~LabelCardinalityReduceExec() on InProcessPlanDispatcher(filodb.core.query.EmptyQueryConfig$@10272bbb)
+             |--E~LabelCardinalityReduceExec() on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1598944375],downsample)
+             |---E~LabelCardinalityExec(shard=0, filters=List(ColumnFilter(_ws_,Equals(ws)), ColumnFilter(_ns_,Equals(ns)), ColumnFilter(_metric_,Equals(metric))), limit=1000000, startMs=1633913330000, endMs=1634755730000) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1598944375],downsample)
+             |---E~LabelCardinalityExec(shard=1, filters=List(ColumnFilter(_ws_,Equals(ws)), ColumnFilter(_ns_,Equals(ns)), ColumnFilter(_metric_,Equals(metric))), limit=1000000, startMs=1633913330000, endMs=1634755730000) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1598944375],downsample)
+             |--E~LabelCardinalityReduceExec() on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1598944375],raw)
+             |---E~LabelCardinalityExec(shard=0, filters=List(ColumnFilter(_ws_,Equals(ws)), ColumnFilter(_ns_,Equals(ns)), ColumnFilter(_metric_,Equals(metric))), limit=1000000, startMs=1634172530000, endMs=1634777330000) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1598944375],raw)
+             |---E~LabelCardinalityExec(shard=1, filters=List(ColumnFilter(_ws_,Equals(ws)), ColumnFilter(_ns_,Equals(ns)), ColumnFilter(_metric_,Equals(metric))), limit=1000000, startMs=1634172530000, endMs=1634777330000) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1598944375],raw)""".stripMargin
+    validatePlan(longTermPlan, expectedLongTermPlan)
+
+    // Case 2: startTime < endTime < rawStart < dsEndTime
```

Comment:
A couple of these look out-of-order (`rawStart < dsEndTime`) but I might just be misunderstanding how this works.

Reply:
No it is correct, ``rawStart`` would be ``now - 7 days`` last ``dsEndTime`` might be ``now - 6 hours ``. First one is the earliest time we have in raw, second one is the latest downsample time.